nym-wallet/src/components/Bonding/modals/BondMixnodeModal.tsx

```diff
@@ -72,28 +77,20 @@ export const BondMixnodeModal = ({
 
   const handleUpdateAmountData = async (data: MixnodeAmount) => {
     setAmountData({ ...data });
+    setStep(3);
+  };
+
+  const handleUpdateSignature = async (data: Signature) => {
+    setSignature(data.signature);
 
     const payload = {
-      pledge: data.amount,
-      ownerSignature: mixnodeData.ownerSignature,
-      mixnode: {
-        ...mixnodeData,
-        mix_port: mixnodeData.mixPort,
-        http_api_port: mixnodeData.httpApiPort,
-        verloc_port: mixnodeData.verlocPort,
-        sphinx_key: mixnodeData.sphinxKey,
-        identity_key: mixnodeData.identityKey,
-      },
-      costParams: {
-        profit_margin_percent: toPercentFloatString(data.profitMargin),
-        interval_operating_cost: {
-          amount: data.operatorCost.amount.toString(),
-          denom: data.operatorCost.denom,
-        },
-      },
+      pledge: amountData.amount,
+      msgSignature: data.signature,
+      mixnode: mixnodeToTauri(mixnodeData),
+      costParams: costParamsToTauri(amountData),
```

Comment:
:ok_hand:

Reply:
Yeah, in the future we can probably refactor these types to avoid that kind of `convertToTheOtherType` logic